source/sync/data-model/update-schema.txt

```diff
@@ -26,443 +21,518 @@ Update Your Data Model
+
+   Several Device SDKs support remapping a property or class name. Depending on
+   your use case, you might choose to remap a name through the client code
+   to avoid having to make a breaking change. For more information, refer to the :ref:`` section 
+
+The following diagram shows the types of changes you can make and the 
+process perform the change:
+
+.. TODO: UPDATE WITH NEW DIAGRAM
+
+.. .. figure:: /images/schema_changes_with_versioning.png
 ..    :alt: Flow chart of schema changes
+..    :lightbox:
+
+Data Model Changes
+------------------ 
+
+The following sections provide more detailed information on how to make
+specific changes to your Device Sync data model.
+
+.. _schema-update-add-object-type:
+
+Add an Object Type
+~~~~~~~~~~~~~~~~~~
+
+You can add an object type to either the server-side schema or the client 
+object model without any additional handling. 
+
+If you want to add an object type to both the server-side schema and the 
+client object model, you can add the object type to the object model, and 
+use :ref:`Development Mode <development-mode>` to let App Services handle the 
+server-side schema updates. Or you can manually add the object type to both 
+the model and the schema.
+
+.. note:: These changes may trigger a resync
+
+   When you add a new object type, we retrieve the documents for the 
+   collection and re-insert them into App Services to get the new values. This
+   is expected behavior, but it does cause a temporary halt to propagating 
+   changes while this process is underway.
+
+.. _schema-update-add-a-default-value:
+
+Add a Default Value on a Property
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add a default value on an object's required property. When you 
+insert an Atlas document missing this required property into the 
+collection, the Device SDK clients set the property to the default value. However, 
+the same property on the Atlas document remains empty until a client makes 
+changes to the property or updates the document directly in Atlas.
+
+For more information on how default values are helpful when modifying 
+existing Atlas documents, see :ref:`<schema-update-add-required-property>`.
+
+.. warning:: Ensure the default value type and property's type are the same
+
+   The default field does not have type validation. If the default field's type
+   and the property's type are not the same, the error will indicate that
+   the document is missing a required field. 
+
+.. _schema-update-add-required-property:
+
+Add a Required Property
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add a required property to the client's object model, and 
+use :ref:`Development Mode <development-mode>` to let App Services infer the 
+server-side schema updates. Or you can manually add the 
+required property to both the client model and the Atlas schema. 
+However, you should consider making the property optional to avoid the need to 
+modify existing Atlas documents. 
```

Comment:
It would be nice to recommend adding a default value when adding a required property to avoid this situation

Reply:
same for this and the following comments: these will be updated (and hopefully, very much improved) as part of the next PR, but this is really helpful feedback. I'll be sure to incorporate it